extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { BookTreeItem } from './bookTreeItem';
+import * as utils from '../common/utils';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+
+export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeItem> {
+
+	private _onDidChangeTreeData: vscode.EventEmitter<BookTreeItem | undefined> = new vscode.EventEmitter<BookTreeItem | undefined>();
+	readonly onDidChangeTreeData: vscode.Event<BookTreeItem | undefined> = this._onDidChangeTreeData.event;
+	private tableOfContentsPath: string[];
+
+	constructor(private workspaceRoot: string) {
+		if (workspaceRoot !== '') {
+			this.tableOfContentsPath = this.getTocFiles(this.workspaceRoot);
+			if (this.tableOfContentsPath === undefined || this.tableOfContentsPath.length === 0) {
```

Comment:
I think code such as below is more readable in this case, but this is just preference.

```TypeScript
let bookOpened: boolean = this.tableOfContentsPath && this.tableOfContentsPath.length > 0;
vscode.commands.executeCommand('setContext', 'bookOpened', bookOpened);
```